stdlib/gc.ml

```diff
@@ -163,3 +163,27 @@ module Memprof =
 
     external discard : t -> unit = "caml_memprof_discard"
   end
+
+
+
+type suspended_collection_work = int
+(* Note: we do not currently expose this type outside the module,
+   because it could plausibly change in the future. In particular,
+   currently the runtime only track major allocations during ramp-up
+   work, but there are other sources of GC pressure, such as custom
+   block allocation, that could be tracked as well and should probably
+   be tracked separately. This suggests that the type of suspended work
+   could become a record of integers instead of one integer.
```

Comment:
In the end, it all boils down to an amount of work that the GC needs to do as a consequence of the allocations, so I doubt it will ever be anything else than an integer.

Reply:
My understanding is that the GC currently tracks three kinds of allocation counters separately: `allocated_words` for in-heap allocations (of which `allocated_words_direct` is a subset), `dependent_allocated` for out-of-heap memory, and `extra_heap_resources` for out-of-heap non-memory resources.

Naively I would expect that the proper way to track dependent and extra-heap allocations during a ramp-up phase is to record them separately from in-heap allocations. In that case the deferred work would be a triple of integers, rather than an integer. I expect whatever representation to remain a field/ring (we can add two amounts, and multiply an amount by a constant (in particular we can split it in a fixed number of almost-equal sub-amounts)).